aws/data_source_aws_imagebuilder_infrastructure_configuration.go

```diff
@@ -0,0 +1,131 @@
+
+		Schema: map[string]*schema.Schema{
+			"arn": {
+				Type:     schema.TypeString,
+				Required: true,
+			},
+			"datecreated": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+			"dateupdated": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+			"description": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+			"instance_profile_name": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+			"instance_types": {
+				Type:     schema.TypeSet,
+				Computed: true,
+				Elem:     &schema.Schema{Type: schema.TypeString},
+				Set:      schema.HashString,
+			},
+			"key_pair": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+			"logging": {
+				Type:     schema.TypeList,
+				Computed: true,
+				Elem: &schema.Resource{
+					Schema: map[string]*schema.Schema{
+						"s3_logs": {
+							Type:     schema.TypeList,
+							Computed: true,
+							Elem: &schema.Resource{
+								Schema: map[string]*schema.Schema{
+									"s3_bucket_name": {
+										Type:     schema.TypeString,
+										Computed: true,
+									},
+									"s3_key_prefix": {
+										Type:     schema.TypeString,
+										Computed: true,
+									},
+								},
+							},
+						},
+					},
+				},
+			},
+			"name": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+			"security_group_ids": {
+				Type:     schema.TypeSet,
+				Computed: true,
+				Elem:     &schema.Schema{Type: schema.TypeString},
+				Set:      schema.HashString,
+			},
+			"sns_topic_arn": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+			"subnet_id": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+			"terminate_instance_on_failure": {
+				Type:     schema.TypeBool,
+				Computed: true,
+			},
+			"tags": tagsSchema(),
+		},
+	}
+}
+
+func dataSourceAwsImageBuilderInfrastructureConfigurationRead(d *schema.ResourceData, meta interface{}) error {
+	conn := meta.(*AWSClient).imagebuilderconn
+	ignoreTagsConfig := meta.(*AWSClient).IgnoreTagsConfig
+
+	componentArn := d.Get("arn").(string)
+
+	params := &imagebuilder.GetInfrastructureConfigurationInput{
+		InfrastructureConfigurationArn: aws.String(componentArn),
+	}
+
+	resp, err := conn.GetInfrastructureConfiguration(params)
+
+	if err != nil {
+		return fmt.Errorf("Error retrieving Component: %s", err)
+	}
+
+	d.SetId(*resp.InfrastructureConfiguration.Arn)
+	d.Set("datecreated", resp.InfrastructureConfiguration.DateCreated)
+	d.Set("dateupdated", resp.InfrastructureConfiguration.DateUpdated)
+	d.Set("date_created", resp.InfrastructureConfiguration.DateCreated)
+	d.Set("description", resp.InfrastructureConfiguration.Description)
+	d.Set("instance_profile_name", resp.InfrastructureConfiguration.InstanceProfileName)
+	d.Set("instance_types", resp.InfrastructureConfiguration.InstanceTypes)
+	d.Set("key_pair", resp.InfrastructureConfiguration.KeyPair)
+	d.Set("logging", resp.InfrastructureConfiguration.Logging)
```

Comment:
It might be better to have the datasource set the logging bucket/prefix as two different outputs, similar to what the [`aws_redshift_cluster` datasource](https://registry.terraform.io/providers/hashicorp/aws/latest/docs/data-sources/redshift_cluster) does

e.g., 

```
d.Set("log_bucket", resp.Infrastructureconfiguration.Logging.S3BucketName)
d.Set("log_prefix", resp.Infrastructureconfiguration.Logging.S3KeyPrefix)
```

Reply:
Yep, @blckct has redone the infrastructure_configuration under #14347

Reply:
Great.  I want to see if there's some way I can help although I've never worked on any TF providers before and have no idea how to get started.

Reply:
Absolutely welcome, this is my first time muddling through! 😄

Reply:
> Absolutely welcome, this is my first time muddling through! 😄

Ok  think I finally figured out that I had messed up my Go installation somehow, and was able to compile and use a custom provider.  let me know what I can do to help

Reply:
Feel free to take your pick from the todo list!
The Image resource looks to be the next easiest big thing that needs doing if you want to have a go on that?

Reply:
Ok.  What is the image resource?  Images are the output created by the image builder pipeline, and would not be managed by Terraform itself.

Reply:
meanwhile working on some lint issues and minor defects - infrastructure config initial create defaults appear different than defaults on update, and some required fields were marked optional in distribution config

Reply:
Yeah I had a look at it, I'm not sure we need the image _resource_, but will need the _datasource_ as the arn is needed for the recipe. It looks like you can have a manual build pipeline with potentially an image resource kicking it off, but I'm not sure how useful that would be..?
I've been a bit naughty and done a couple of rebased/force pushes this week, as after finishing the pipeline and trying it there were some issues (especially with the distribution config resource!) so apologies if that's messed you up!